src/pip/_internal/network/download.py

```diff
@@ -34,48 +42,48 @@ def _get_http_response_size(resp):
         return None
 
 
-def _prepare_download(
-    resp,  # type: Response
-    link,  # type: Link
-    progress_bar  # type: str
-):
-    # type: (...) -> Iterable[bytes]
-    total_length = _get_http_response_size(resp)
-
+def _log_response(response_size, from_cache, link):
+    # type: (Optional[int], bool, Link) -> None
     if link.netloc == PyPI.file_storage_domain:
-        url = link.show_url
+        url = redact_auth_from_url(link.show_url)
     else:
-        url = link.url_without_fragment
-
-    logged_url = redact_auth_from_url(url)
+        url = redact_auth_from_url(link.url_without_fragment)
 
-    if total_length:
-        logged_url = '{} ({})'.format(logged_url, format_size(total_length))
+    if response_size is None:
+        url_repr = '{}'
+    else:
+        url_repr = '{{}} ({})'.format(format_size(response_size))
+    logged_url = url_repr.format(url)
```

Comment:
I personally find this `url_repr` indirection somewhat hard to follow. Is there a reason we're not just doing:
```python
if response_size is None:
    logged_url = url
else:
    logged_url = '{} ({})'.format(url, format_size(response_size))
```

Reply:
I've just addressed this, thanks!